src/m365/file/commands/file-move.ts

```diff
@@ -0,0 +1,240 @@
+
+    const response: any = isPatch ? await request.patch(requestOptions) : await request.post(requestOptions);
+
+    if (isPatch) {
+      return response;
+    }
+    else {
+      await this.waitUntilCopyOperationCompleted(response.headers.location, logger);
+      return response;
+    }
+  }
+
+  private getAbsoluteUrl(webUrl: string, url: string): string {
+    return url.startsWith('https://') ? url : urlUtil.getAbsoluteUrl(webUrl, url);
+  }
+
+  private async getDriveIdAndItemId(webUrl: string, folderUrl: string, sourceUrl: string, logger: Logger, verbose?: boolean): Promise<{ siteId: string, driveId: string, drive: Drive, itemId: string }> {
+    const siteId: string = await spo.getSiteId(webUrl, logger, verbose);
+    const drive: Drive = await this.getDocumentLibrary(siteId, new URL(folderUrl), sourceUrl);
+    const itemId: string = await this.getStartingFolderId(drive, new URL(folderUrl));
+    return { siteId, driveId: drive.id as string, drive, itemId };
+  }
+
+  private async getDocumentLibrary(siteId: string, folderUrl: URL, folderUrlFromUser: string): Promise<Drive> {
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/sites/${siteId}/drives?$select=webUrl,id`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const drives = await request.get<{ value: Drive[] }>(requestOptions);
+    const lowerCaseFolderUrl: string = folderUrl.href.toLowerCase();
+
+    const drive: Drive | undefined = drives.value
+      .sort((a, b) => (b.webUrl as string).localeCompare(a.webUrl as string))
+      .find((d: Drive) => {
+        const driveUrl: string = (d.webUrl as string).toLowerCase();
+        // ensure that the drive url is a prefix of the folder url
+        return lowerCaseFolderUrl.startsWith(driveUrl) &&
+          (driveUrl.length === lowerCaseFolderUrl.length ||
+            lowerCaseFolderUrl[driveUrl.length] === '/');
+      });
+
+    if (!drive) {
+      throw `Document library '${folderUrlFromUser}' not found`;
+    }
+
+    return drive;
+  }
+
+  private async getStartingFolderId(documentLibrary: Drive, folderUrl: URL): Promise<string> {
+    const documentLibraryRelativeFolderUrl: string = folderUrl.href.replace(new RegExp(`${documentLibrary.webUrl}`, 'i'), '').replace(/\/+$/, '');
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/drives/${documentLibrary.id}/root${documentLibraryRelativeFolderUrl ? `:${documentLibraryRelativeFolderUrl}` : ''}?$select=id`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const folder = await request.get<DriveItem>(requestOptions);
+
+    return folder?.id as string;
+  }
+
+  private async getTargetDriveAndItemId(webUrl: string, targetUrl: string, logger: Logger, verbose?: boolean): Promise<{ targetDriveId: string, targetItemId: string }> {
```

Comment:
I think this function is a bit of an overkill. Let's move the first two lines to the commandAction method, and let's just reuse the `getDriveIdAndItemId` function to get the information of the target